lib/mongo/srv/records.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2014-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'resolv'
+
+module Mongo
+
+  module SRV
+
+    # This SRV::Records class is used to keep track of the SRV records discovered for a given hostname. It also keeps
+    # track of the minimum TTL of the records added so far.
+    #
+    # @api private
+    class Records
+
+      MISMATCHED_DOMAINNAME = "Parent domain name in SRV record result (%s) does not match " +
+                                 "that of the hostname (%s)".freeze
+
+      attr_reader :hostname
```

Comment:
Even though this entire class is private, please add docstrings with types for each attribute and a description of what they store as well as docstrings for the methods with descriptions of what they do and types for their parameters.